website/docs/components/code-block/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,93 @@
+## Usage
+
+### When to use
+
+- When displaying code examples and longer snippets of code that benefit from syntax highlighting.
+
+### When not to use
+
+- As a full-blown code editor.
+
+### When to use a CopySnippet vs a CodeBlock
+
+There is some overlap in the copying functionality of the CopySnippet and the CodeBlock. Which to use generally comes down to the complexity of the code/value displayed within the component, and whether the user benefits from seeing the larger context of the code example.
+
+For example; if allowing the user to copy an API key or other **singular** value where they don’t necessarily need to see the entire context of where the copied value is expected to be pasted into, use a CopySnippet.
+
+Use a CodeBlock if:
+
+- The user benefits from seeing the larger context of where a value should be pasted into, e.g., in a configuration file.
```

Comment:
I am not sure the code-block allows to "seeing the larger context of where a value should be pasted into, e.g., in a configuration file".

For me the main difference is this one:  the main action for a code block is seeing the code, the copy is secondary; for a copy snippet is the other way round.

Reply:
That's a great breakdown, thanks for that!